lib/job.js

```diff
@@ -368,14 +368,14 @@ Job.prototype.finished = function(){
         //
         interval = setInterval(function(){
           scripts.isFinished(_this).then(function(status){
-            var finished = status > 0;
-            if(finished){
+            Job.fromId(_this.queue, _this.id).then(function(job){
+              removeListeners();
               if(status == 2){
-                getFailed().then(resolve, reject);
+                reject(new Error(job.failedReason));
+              } else {
+                resolve(job.returnvalue); 
               }
-              removeListeners();
-              resolve();
-            }
+            });
```

Comment:
Sorry, I misunderstood isFinished, I thought it only resolve when job status is already finished. Will fix this, tks.